### PR TITLE
feat: Constructing collections with builder functions

### PR DESCRIPTION
**Pull request by danil-pavlov**

### CODE REVIEW

**Comment by ilya-g** [comment]
There are still many unrelated commits in this branch. Please try rebasing it on top of the target branch: `1-6-doc-update`.
Currently it seems to be merged with that branch, and not rebased onto it.

File: docs/topics/constructing-collections.md [hunk 1]
```diff
@@ -31,6 +31,24 @@ initialization fluent here.
 val numbersMap = mutableMapOf<String, String>().apply { this["one"] = "1"; this["two"] = "2" }
 ```
 
+## Create with collection builder functions
+
+Another way of creating a collection is to call a builder function –
+[`buildList()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/build-list.html), [`buildSet()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/build-set.html),
+or [`buildMap()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/build-map.html). They create a new
```
Comment (koshachy) [inline docs/topics/constructing-collections.md:38]: I feel awkward whenever I use pronouns for such objects — functions, classes, etc.
I prefer to use "These functions create". But maybe it looks redundant. 
WDYT?

File: docs/topics/constructing-collections.md [hunk 2]
```diff
@@ -31,6 +31,24 @@ initialization fluent here.
...
+mutable collection of the corresponding type, populate it using [write operations](collection-write.md),
+and return a read-only collection with the same elements.
```
Comment (koshachy) [inline docs/topics/constructing-collections.md:40]: It's better to use a colon if the following code block shows an example.

**Review by koshachy — APPROVED** [review]
Looks good.
Start a discussion if you have any questions or comments.

File: docs/topics/constructing-collections.md (hunk 1 above)
Comment (danil-pavlov) [reply docs/topics/constructing-collections.md:38]: I think it's OK to use such pronouns even with inanimate objects. It looks simpler and more colloquial
I can double-check with copyeditors

File: docs/topics/constructing-collections.md (hunk 2 above)
Comment (danil-pavlov) [reply docs/topics/constructing-collections.md]: Thanks, fixed

**Pull request merged**